pkg/aws/ec2.go

```diff
@@ -0,0 +1,62 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/smithy-go"
+)
+
+type EC2Client interface {
+	DescribeImages(ctx context.Context, params *ec2.DescribeImagesInput, optFns ...func(*ec2.Options)) (*ec2.DescribeImagesOutput, error)
+	DescribeKeyPairs(ctx context.Context, params *ec2.DescribeKeyPairsInput, optFns ...func(*ec2.Options)) (*ec2.DescribeKeyPairsOutput, error)
+	CreateKeyPair(ctx context.Context, params *ec2.CreateKeyPairInput, optFns ...func(*ec2.Options)) (*ec2.CreateKeyPairOutput, error)
+}
+
+func NewEC2Client(config aws.Config) *ec2.Client {
+	return ec2.NewFromConfig(config)
+}
+
+func (a *client) ImageExists(ctx context.Context, imageID string) (bool, error) {
```

Comment:
Using `a` as the receiver parameter which doesn't seem to be related to `*client` in any capacity.